framework/platform/window.h

```diff
@@ -17,27 +17,69 @@
 
 #pragma once
 
+#include "common/optional.h"
 #include "common/vk_common.h"
 #include "core/instance.h"
 #include "vulkan/vulkan.hpp"
 
 namespace vkb
 {
-class Platform;
-
 /**
  * @brief An interface class, declaring the behaviour of a Window
  */
 class Window
 {
   public:
+	struct Extent
+	{
+		uint32_t width;
+		uint32_t height;
+	};
+
+	struct OptionalExtent
+	{
+		Optional<uint32_t> width;
+		Optional<uint32_t> height;
+	};
+
+	enum class Mode
+	{
+		Headless,
+		Fullscreen,
+		FullscreenBorderless,
+		Default
+	};
+
+	enum class Vsync
+	{
+		OFF,
+		ON,
+		Default
+	};
+
+	struct OptionalProperties
+	{
+		Optional<std::string> title;
+		Optional<Mode>        mode;
+		Optional<bool>        resizable;
+		Optional<Vsync>       vsync;
+		OptionalExtent        extent;
+	};
+
+	struct Properties
+	{
+		std::string title     = "";
+		Mode        mode      = Mode::Default;
+		bool        resizable = true;
+		Vsync       vsync     = Vsync::Default;
+		Extent      extent    = {1024, 768};
```

Comment:
With this default extent, all samples start with a 4:3 ratio, which differs from what's currently used in master (1280x760, 16:6).